src/briefcase/platforms/web/static.py

```diff
@@ -0,0 +1,347 @@
+            # Trim previously compiled content out of briefcase.css
+            briefcase_css_path = (
+                self.project_path(app) / "static" / "css" / "briefcase.css"
+            )
+            self._trim_file(
+                briefcase_css_path,
+                sentinel=" ******************* Wheel contributed styles **********************/",
+            )
+
+            # Extract static resources from packaged wheels
+            for wheelfile in self.wheel_path(app).glob("*.whl"):
+                self.logger.info(f"Processing {wheelfile.name}...")
+                with briefcase_css_path.open("a", encoding="utf-8") as css_file:
+                    self._process_wheel(wheelfile, css_file=css_file)
+
+        return {}
+
+
+class HTTPHandler(SimpleHTTPRequestHandler):
+    """Convert any HTTP request into a path request on the static content
+    folder."""
+
+    def translate_path(self, path):
+        return str(self.server.base_path / path[1:])
+
+
+class LocalHTTPServer(HTTPServer):
+    """A HTTP server that serves local static content."""
+
+    def __init__(self, base_path, host, port, RequestHandlerClass=HTTPHandler):
+        self.base_path = base_path
+        super().__init__((host, port), RequestHandlerClass)
+
+
+class StaticWebRunCommand(StaticWebMixin, RunCommand):
+    description = "Run a static web project."
+
+    def add_options(self, parser):
+        super().add_options(parser)
+        parser.add_argument(
+            "--host",
+            default="localhost",
+            help="The host on which to run the server (default: localhost)",
+            required=False,
+        )
+        parser.add_argument(
+            "-p",
+            "--port",
+            default=8080,
+            type=int,
+            help="The port on which to run the server (default: 8080)",
+            required=False,
+        )
+        parser.add_argument(
+            "--no-browser",
+            action="store_false",
+            dest="open_browser",
+            help="Don't open a web browser on the newly opened server.",
+            required=False,
+        )
+
+    def run_app(self, app: AppConfig, host, port, open_browser, **kwargs):
+        """Start the application.
+
+        :param app: The config object for the app
+        :param host: The host on which to run the server
+        :param port: The device UDID to target. If ``None``, the user will
+            be asked to select a device at runtime.
+        :param open_browser: Should a browser be opened on the newly started
+            server.
+        """
+        self.logger.info(
+            f"Starting web server on http://{host}:{port}/",
+            prefix=app.app_name,
+        )
+
+        httpd = None
+        try:
+            # Create a local HTTP server
+            httpd = LocalHTTPServer(
+                self.project_path(app),
+                host=host,
+                port=port,
+            )
```

Comment:
curious if this was intentional design or not. it is pretty easy to get this to fail; for instance, just specify a privileged port like 80.

Should we catch these exceptions or just give them to the user?

Reply:
It was only intentional in the sense of "don't try and be too smart about it". I've added explicit handling and an error message for the permission error raised by low ports.

Reply:
> I don't actually know what this one does... 🙃

Oh no... you've opened a connection to the NETHERWORLD!!! :-)

I've added more common error cases.

Reply:
Another path to the netherworld.... 👻 

```
❯ briefcase run web -p 0

[webtemplate] Starting web server on http://localhost:0/

[webtemplate] Web server log output (type CTRL-C to stop log)...
===========================================================================
```

Oh....interesting....this actually asks the OS to use a random available port....cool.

Reply:
Ah - that one is legal - and points out that we shouldn't be just reflecting the port from the input arguments, we should be using `getsockname()`.